lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -424,10 +426,15 @@ def ipv4_address(vm)
     # Multiple reboots occur during guest customization in which a link-local
     # address is assigned. As such, we need to wait until a routable IP address
     # becomes available. This is most commonly an issue with Windows instances.
-    sleep 2 while vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer'
+    sleep 2 while vm_is_waiting_for_ip?(vm)
     vm.guest.net[bootstrap_nic_index].ipAddress.detect { |addr| IPAddr.new(addr).ipv4? }
   end
 
+  def vm_is_waiting_for_ip?(vm)
+    first_ip_address = vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }
+    first_ip_address.nil? || first_ip_address == ORIGIN_IS_REAL_NIC
```

Comment:
I probably led you astray on this... I think the second half of the OR clause should refer to `first_ip_address.origin` and not the object itself.